crates/driver/src/logic/eth/mod.rs

```diff
@@ -187,8 +248,18 @@ pub struct Interaction {
 /// An onchain transaction.
 #[derive(Debug)]
 pub struct Tx {
-    pub from: Address,
+    pub from: Account,
     pub to: Address,
     pub value: Ether,
     pub input: Vec<u8>,
+    pub access_list: AccessList,
```

Comment:
Same, why not `Option<AccessList>` since empty access list is still a valid list.

Reply:
I think we will send the tx onchain without access list if
1. Access list estimators failed for any reason (we should not abort the submission process if we don't have access list)
2. Access list estimators disabled by configuration (in cases of high percent of failed/reverted transactions onchain)

On the other hand, I _think_ that nothing will be different if you send a tx with empty access list compared to tx without access list, at least for the node clients (although I did not try it). But we do have 3rd party submitters (infura, nodereal) and I can only guess how they treat it.

Reply:
They are equivalent AFAIU - because `None` get encoded as an empty access list in EIP-1559 transactions anyway.